share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
Could it be that Monty is in fact including the system `pthread.h` headers when he writes `#include <pthread.h>` — and not his own custom `pthread.h` header? I suspect so. That would also explain the order dependency on `#include "pthreadtypes.h"`. If this is the case, then his implementation actually depends on a pthread API implementation which has a file called `pthreadtypes.h` with certain properties. This seems to be very platform dependent.

Note: this is what happens on OS X:
````
$ find /usr/include -name "pthread.h"
/usr/include/pthread/pthread.h
/usr/include/pthread.h
$ find /usr/include -name "pthreadtypes.h"
````
Which means that the current solution is not portable.

If we were simply to ensure that `#include <pthread.h>` pulled in Monty’s version of `pthread.h`, then we would have a clean, portable solution. And no need for this non-portable `pthreadtypes.h` stuff.